lib/matplotlib/backends/_backend_pdf_ps.py

```diff
@@ -107,23 +107,98 @@ class CharacterTracker:
     """
     Helper for font subsetting by the PDF and PS backends.
 
-    Maintains a mapping of font paths to the set of glyphs that are being used from that
-    font.
-    """
+    Maintains a mapping of font paths to the set of characters and glyphs that are being
+    used from that font.
 
-    def __init__(self) -> None:
-        self.used: dict[str, set[GlyphIndexType]] = {}
+    Attributes
+    ----------
+    subset_size : int
+        The size at which characters are grouped into subsets.
+    used : dict[tuple[str, int], dict[CharacterCodeType, GlyphIndexType]]
+        A dictionary of font files to character maps. The key is a font filename and
+        subset within that font. The value is a dictionary mapping a character code to a
+        glyph index. If *subset_size* is not set, then there will only be one subset per
+        font filename.
+    """
 
-    def track(self, font: FT2Font, s: str) -> None:
-        """Record that string *s* is being typeset using font *font*."""
+    def __init__(self, subset_size: int = 0):
+        """
+        Parameters
+        ----------
+        subset_size : int, optional
+            The maximum size that is supported for an embedded font. If provided, then
+            characters will be grouped into these sized subsets.
+        """
+        self.used: dict[tuple[str, int], dict[CharacterCodeType, GlyphIndexType]] = {}
```

Comment:
Your definition of "charmap" is a map of charcodes to glyph indices, whereas freetype's definition maps glyph indices to charcodes (see e.g. the swapping in embedTTFType3.  Both are obviously equivalent and I can imaging why this definition actually works better for us, but that probably warrants a comment?